crates/hotshot/examples/infra/mod.rs

```diff
@@ -101,20 +108,18 @@ pub fn load_config_from_file<TYPES: NodeType>(
     let mut config: NetworkConfig<TYPES::SignatureKey, TYPES::ElectionConfigType> =
         config_toml.into();
 
-    // Generate network's public keys
-    let known_nodes: Vec<_> = (0..config.config.total_nodes.get())
-        .map(|node_id| {
-            TYPES::SignatureKey::generated_from_seed_indexed(
-                config.seed,
-                node_id.try_into().unwrap(),
-            )
-            .0
-        })
-        .collect();
-
-    config.config.known_nodes_with_stake = (0..config.config.total_nodes.get())
-        .map(|node_id| known_nodes[node_id].get_stake_table_entry(1u64))
-        .collect();
+    // my_own_validator_config would be best to load from file,
+    // but its type is too complex to load so we'll generate it from seed now
+    config.config.my_own_validator_config =
+        ValidatorConfig::generated_from_seed_indexed(config.seed, config.node_index, 1);
+    let my_own_validator_config_with_stake = config
+        .config
+        .my_own_validator_config
+        .public_key
+        .get_stake_table_entry(1u64);
+    // initialize it with size for better assignment of other peers' config
+    config.config.known_nodes_with_stake =
+        vec![my_own_validator_config_with_stake; config.config.total_nodes.get() as usize];
```

Comment:
In the previous version, we generated different keys for peers, but now it looks like they are all identical. Is this intentional?

Reply:
From the comment on this function, this is indeed intentional. Is it because we can only generate the key for ourselves, and wait for the orchestrator to send peers' keys to us?

Reply:
Yes this is intentional, basically for initializing a vector with fixed type and size! So that when we assign peer's public key to the config, we can directly do `vec[index] = ...` which is easier.